### PR TITLE
[EVM] add endpoint to send transaction

### PR DESCRIPTION
**Pull request by codchen**
## Describe your changes and provide context
Implement `SendRawTransaction`. Note that for now Sei nodes will not custody users' private keys so it will not offer endpoints that provide signing capability (e.g. SendTransaction, Sign, SignTransaction)

## Testing performed to validate your change
unit tests

### CODE REVIEW

**Comment by jewei1997** [comment]
nit: can possibly remove this: https://github.com/sei-protocol/sei-chain/blob/5c4b9c82f0272c012e1bdbcbded87088f0b49521/evmrpc/testutils.go#L243

File: evmrpc/send_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	ethCfg := evmParams.GetChainConfig().EthereumConfig(big.NewInt(1))
+	signer := ethtypes.MakeSigner(ethCfg, big.NewInt(Ctx.BlockHeight()), uint64(Ctx.BlockTime().Unix()))
+	tx := ethtypes.NewTx(&txData)
+	tx, err := ethtypes.SignTx(tx, signer, key)
+	require.Nil(t, err)
+	bz, err := tx.MarshalBinary()
+	require.Nil(t, err)
+	payload := "0x" + hex.EncodeToString(bz)
+
+	resObj := sendRequestGood(t, "sendRawTransaction", payload)
```
Comment (jewei1997) [inline evmrpc/send_test.go:42]: is there no checking on the validity of the transaction currently? I changed the nonce to some random number for example and the transaction still worked.

File: evmrpc/send.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	}
+	txBuilder := s.txConfig.NewTxBuilder()
+	if err := txBuilder.SetMsgs(msg); err != nil {
+		return common.Hash{}, err
+	}
+	txbz, err := s.txConfig.TxEncoder()(txBuilder.GetTx())
+	if err != nil {
+		return common.Hash{}, err
+	}
+	res, err := s.tmClient.BroadcastTxSync(ctx, txbz)
```
Comment (jewei1997) [inline evmrpc/send.go]: why are we using `BroadcastTxSync` over `BroadcastTx`? What is the difference? It says `BroadcastTxSync` is deprecated. https://github.com/sei-protocol/sei-tendermint/blob/962ce074d5dd07c2ff9a3e81b94755c18481774c/rpc/client/interface.go#L66-L69

File: evmrpc/send_test.go (hunk 1 above)
Comment (codchen) [reply evmrpc/send_test.go:42]: the check happens in a real tendermint client, which will call CheckTx on the transaction https://github.com/sei-protocol/sei-chain/pull/1109

In the unit test we use a mock so there is no check

File: evmrpc/send.go (hunk 2 above)
Comment (codchen) [reply evmrpc/send.go]: these two are the same: https://github.com/sei-protocol/sei-tendermint/blob/main/internal/rpc/core/mempool.go#L32-L34 I've changed to use BroadcastTx instead

**Comment by codchen** [comment]
> nit: can possibly remove this:
> 
> https://github.com/sei-protocol/sei-chain/blob/5c4b9c82f0272c012e1bdbcbded87088f0b49521/evmrpc/testutils.go#L243

@jewei1997 i think linter would exclude reference from test files and still consider as this deadcode if we remove this nolint :(

**Review by jewei1997 — APPROVED** [review]

**Review by philipsu522 — APPROVED** [review]

**Pull request merged**